src/libsyntax/json.rs

```diff
@@ -170,6 +171,27 @@ impl Diagnostic {
                 rendered: None,
             }
         });
+
+        // generate regular command line output and store it in the json
+
+        // A threadsafe buffer for writing.
+        #[derive(Default, Clone)]
```

Comment:
To clarify, this is needed only because the interface of `EmitterWriter::new` requires it?

Reply:
Yes